ipywidgets/static/widgets/js/manager.js

```diff
@@ -317,6 +317,12 @@ define([
         // Recreate all the widget models for the given notebook state.
         var all_models = that._get_comm_info().then(function(live_comms) {
             return Promise.all(_.map(Object.keys(state), function (model_id) {
+                
+                // If the model has already been created, return it.
+                if (that._models[model_id]) {
+                    return that._models[model_id];
+                }
+                
```

Comment:
In line 425 of the block below.  Before I call this function I create models based on `_get_comm_info` return info.